src/Bridge/Doctrine/Orm/Filter/AbstractFilter.php

```diff
@@ -283,12 +283,24 @@ protected function addJoinsForNestedProperty(string $property, string $rootAlias
             $resourceClass = null;
         }
 
+        if (\func_num_args() > 5) {
+            $joinType = func_get_arg(5);
+        } else {
+            if (__CLASS__ !== \get_class($this)) {
+                $r = new \ReflectionMethod($this, __FUNCTION__);
+                if (__CLASS__ !== $r->getDeclaringClass()->getName()) {
+                    @trigger_error(sprintf('Method %s() will have a sixth `$joinType` argument in version API Platform 3.0. Not defining it is deprecated since API Platform 2.3.', __FUNCTION__), E_USER_DEPRECATED);
+                }
+            }
+            $joinType = null;
+        }
+
         $propertyParts = $this->splitPropertyParts($property, $resourceClass);
         $parentAlias = $rootAlias;
         $alias = null;
 
         foreach ($propertyParts['associations'] as $association) {
-            $alias = QueryBuilderHelper::addJoinOnce($queryBuilder, $queryNameGenerator, $parentAlias, $association);
+            $alias = QueryBuilderHelper::addJoinOnce($queryBuilder, $queryNameGenerator, $parentAlias, $association, $joinType);
```

Comment:
If we use another filter before `OrderFilter` on the same field, the join type will be `INNER`.

Reply:
Which is correct. But for such cases, we need a way to replace the join with an inner join when there was a previous left join.

Reply:
Definitely right, I'll try to fix this as well.